types/src/addressable_entity.rs

```diff
@@ -608,6 +609,113 @@ impl KeyValueJsonSchema for EntryPointLabels {
     const JSON_SCHEMA_KV_NAME: Option<&'static str> = Some("NamedEntryPoint");
 }
 
+/// Collection of named message topics.
+#[derive(Clone, PartialEq, Eq, Serialize, Deserialize, Debug, Default)]
+#[cfg_attr(feature = "datasize", derive(DataSize))]
+#[cfg_attr(feature = "json-schema", derive(JsonSchema))]
+#[serde(transparent, deny_unknown_fields)]
+pub struct MessageTopics(
+    #[serde(with = "BTreeMapToArray::<String, TopicNameHash, MessageTopicLabels>")]
+    BTreeMap<String, TopicNameHash>,
+);
+
+impl ToBytes for MessageTopics {
```

Comment:
Shall we add the serialization roundtrip test for `MessageTopics`?

Reply:
added here: https://github.com/casper-network/casper-node/pull/4315/commits/3190647ec0012448907208e3a8406244433c5486